autoload/go/complete.vim

```diff
@@ -19,13 +19,6 @@ function! s:gocodeCommand(cmd, preargs, args) abort
     return
   endif
 
-  " we might hit cache problems, as gocode doesn't handle well different
-  " GOPATHS: https://github.com/nsf/gocode/issues/239
-  let old_gopath = $GOPATH
-  let old_goroot = $GOROOT
-  let $GOPATH = go#path#Detect()
-  let $GOROOT = go#util#env("goroot")
```

Comment:
Not 100% sure about removing this particular case though. Need to carefully read that issue.

Reply:
this is needed for Go upgrades. Unfortunately `gocode` embeds the goroot into the binary instead of looking from env, after a Go upgrade `gocode` can be broken if the user has a `version` based path. i.e: `/usr/lib/go/1.9.1/src`.

Reply:
We should keep the goroot behavior but remove the gopath change.